src/runtime/pkg/hypervisors/hypervisor_state.go

```diff
@@ -26,6 +28,32 @@ type CPUDevice struct {
 	ID string
 }
 
+// PCIePort distinguish only between root and switch port
+type PCIePort string
+
+const (
+	// RootPort attach VFIO devices to a root-port
+	RootPort PCIePort = "root-port"
+	// SwitchPort attach VFIO devices to a switch-port
+	SwitchPort = "switch-port"
+	// BridgePort is the default
+	BridgePort = "bridge-port"
+	// NoPort is for disabling VFIO hotplug/coldplug
+	NoPort = "no-port"
+)
+
+func (p PCIePort) String() string {
+	switch p {
+	case RootPort:
+		return "root-port"
+	case SwitchPort:
+		return "switch-port"
+	case NoPort:
+		return "no-port"
+	}
```

Comment:
```suggestion
	switch p {
	case RootPort:
	case SwitchPort:
	case NoPort:
		return string(p)
	}
```

Reply:
Great catch, I even forgot to add the BridgePort that is far more conveient.